### PR TITLE
feat: add EIP4844 transactions

### LINKED ISSUE #26: Add support for validating EIP-4844 transactions
https://eips.ethereum.org/EIPS/eip-4844

### PR DESCRIPTION
**Pull request by fmoletta**
This PR is based on #138  (as they modify the same code), please merge it first

**Motivation**

Add support for EIP4844 transactions

**Description**

Add `EIP4844` transactions + needed trait and methods and integrate it into existing transaction-related code

Closes #26 (evm already performs validations before executing so adding the transaction itself will sufice)

### CODE REVIEW

File: crates/core/src/rlp/error.rs [hunk 1]
```diff
@@ -13,6 +13,8 @@ pub enum RLPDecodeError {
     UnexpectedList,
     #[error("UnexpectedString")]
     UnexpectedString,
+    #[error("Invalid transaction type")]
+    InvalidTransactionType,
```
Comment (MegaRedHand) [inline crates/core/src/rlp/error.rs:17]: This doesn't belong here.

File: crates/core/src/types/engine/payload.rs [hunk 2]
```diff
@@ -73,11 +73,19 @@ impl EncodedTransaction {
...
                     // EIP1559
                     0x2 => {
                         EIP1559Transaction::decode(tx_bytes).map(Transaction::EIP1559Transaction)
                     }
-                    _ => unimplemented!("We don't know this tx type yet"),
+                    // EIP4844
+                    0x3 => {
+                        EIP4844Transaction::decode(tx_bytes).map(Transaction::EIP4844Transaction)
+                    }
+                    _ => Err(RLPDecodeError::InvalidTransactionType),
```
Comment (MegaRedHand) [inline crates/core/src/types/engine/payload.rs:88]: Can we change the return type, or return `MalformedData` instead?

Comment (fmoletta) [reply crates/core/src/types/engine/payload.rs:88]: Could we use the custom variant? It is a shame to lose key information useful for debugging

Comment (MegaRedHand) [reply crates/core/src/types/engine/payload.rs]: Sure!

**Pull request merged**